tests/test_modeling_blenderbot.py

```diff
@@ -0,0 +1,426 @@
+    dropout=self.hidden_dropout_prob,
+    vocab_size=self.vocab_size,
+    encoder_layers=self.num_hidden_layers,
+    decoder_layers=self.num_hidden_layers,
+    encoder_attention_heads=self.num_attention_heads,
+    decoder_attention_heads=self.num_attention_heads,
+    attention_dropout=self.attention_dropout,
+    encoder_ffn_dim=self.ffn_size,
+    decoder_ffn_dim=self.ffn_size,
+    max_position_embeddings=self.max_position_embeddings,
+    variant=self.variant,
+    scale_embedding=self.scale_embedding,
+    bos_token_id=self.bos_token_id,
+    eos_token_id=self.eos_token_id,
+    pad_token_id=self.pad_token_id,
+    num_beams=1,
+    min_length=3,
+    max_length=10,
+    )
+    attention_mask = ids_tensor([self.batch_size, self.seq_len], vocab_size=2)
+    inputs_dict = {"input_ids": input_ids, "attention_mask": attention_mask}
+    return config, inputs_dict
+
+
+@require_torch
+class BlenderbotTesterMixin(ModelTesterMixin, unittest.TestCase):
+ all_generative_model_classes = (BlenderbotForConditionalGeneration,) if is_torch_available else ()
+
+ is_encoder_decoder = True
+ test_head_masking = False
+ test_pruning = False
+ test_torchscript = True
+ test_resize_embeddings = True
+ test_missing_keys = False
+
+ def setUp(self):
+  self.model_tester = BlenderbotModelTester(self)
+  self.config_tester = ConfigTester(self, config_class=BlenderbotConfig)
+
+ def test_inputs_embeds(self):
+  pass
+
+ def test_initialization_module(self):
+  config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+  model = BlenderbotForConditionalGeneration(config)
+  model.to(torch_device)
+  model.eval()
+  self.assertTrue((model.encoder.embed_tokens.weight == model.shared.weight).all().item())
+  self.assertAlmostEqual(torch.std(model.encoder.embed_tokens.weight).item(), config.init_std, 2)
+  self.assertAlmostEqual(torch.std(model.encoder.embed_positions.weight).item(), config.init_std, 2)
+
+ def test_embed_pos_shape(self):
+  config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+  model = BlenderbotForConditionalGeneration(config)
+  expected_shape = (config.max_position_embeddings, config.d_model)
+  self.assertEqual(model.encoder.embed_positions.weight.shape, expected_shape)
+  self.assertEqual(model.decoder.embed_positions.weight.shape, expected_shape)
+
+
+@require_torch
+class AbstractBlenderBotIntegrationTests(unittest.TestCase):
+ checkpoint_name = "sshleifer/blenderbot-3B"
+ tokenizer_cls = BlenderbotTokenizer
+
+ @cached_property
+ def model(self):
+  model = BlenderbotForConditionalGeneration.from_pretrained(self.checkpoint_name).to(torch_device)
+  if torch_device == "cuda":
+   model = model.half()
+  return model
+
+ @cached_property
+ def tokenizer(self):
+  return self.tokenizer_cls.from_pretrained(self.checkpoint_name)
+
+
+#@unittest.skipUnless(torch_device != "cpu", "3b test are very slow on CPU.")
+@require_torch
+class Blenderbot3BIntegrationTests(AbstractBlenderBotIntegrationTests):
```

Comment:
do these pass?
I would still skip them on CPU and mark them slow.

Reply:
Yes it's working